.github/workflows/code-quality-checks.yml

```diff
@@ -0,0 +1,85 @@
+name: Checks
```

Comment:
There's a benefit of having the checks in individual jobs instead of a single job. You can see all results individually directly in the PR and don't have to check the steps of the job. This also means that if the formatting fails you also see if the tests would pass and can fix both at the same time rather than fixing tests, pushing and then only seeing that the tests fail, as the checks are run in parallel rather than series.

Regarding efficiency, why do you think it's a problem that it is a bit inefficient? The jobs are executed in parallel, so it should not take that much longer than using only a single job, even if the setup is done multiple times. Maybe the `npm ci` can be done in a shared job before, but I'm not sure if it's worth it. I personally think the benefits of having multiple jobs outweigh the gains of efficiency of using a single job.

Reply:
Well, that is a good argument. And the performance is not really an issue in this case because we're an OpenSource project on Github and don't have to pay anything for the actions. I'd say we'll leave it as it currently is 👌